src/metabase/lib/expression.cljc

```diff
@@ -280,3 +280,39 @@
 (defmethod lib.ref/ref-method :expression
   [expression-clause]
   expression-clause)
+
+(mu/defn expressionable-columns :- [:sequential lib.metadata/ColumnMetadata]
+  "Get column metadata for all the columns that can be used expressions in
+  the stage number `stage-number` of the query `query` and in expression index `expression-position`
+  If `stage-number` is omitted, the last stage is used.
+  Pass nil to `expression-position` for new expressions.
+  The rules for determining which columns can be broken out by are as follows:
+
+  1. custom `:expressions` in this stage of the query, that come before the `expression-position`
+
+  2. Fields 'exported' by the previous stage of the query, if there is one;
+     otherwise Fields from the current `:source-table`
+
+  3. Fields exported by explicit joins
+
+  4. Fields in Tables that are implicitly joinable."
```

Comment:
I don't know if it's really valuable to be parroting the definition of `visible-columns` every time we write a function that uses it, I'm thinking instead of copying this from `filterable-columns` we should actually just delete it from there and make it part of the docstring for `visible-columns` and point people to that.